### ISSUE TITLE
Vyatta/VyOS backup

### ISSUE THREAD

**Issue by sallysandtrap** [ytti/oxidized#840]
I'm new to GitHub from a posting perspective so I apologize if this should be posted somewhere other than issues as this is more of a recommendation.

For the vyatta.rb class you are issuing "show configuration | no-more" to output the running configuration in it's true form.  While this is great to have it is also very beneficial on this platform to include "show configuration commands | no-more" as part of the backup.  Similar to the output of a "show run" for Cisco configurations, "show configuration commands" will output the actual commands a user would need to enter to achieve the running configuration.

In other words, if my VyOS router dies and I stood up a new one, I could simply copy and paste the "show configuration commands" output and get it back up-and-running.

I modified the vyatta.rb in my installation and it works fine on a VyOS 1.1.7 (helium) platform:

class Vyatta < Oxidized::Model

  # Brocade Vyatta / VyOS model #

  prompt /\@.*?\:~\$\s/

  cmd :all do |cfg|
    cfg = cfg.lines.to_a[1..-2].join
  end

  cmd :secret do |cfg|
    cfg.gsub! /community (\S+) {/, 'community <hidden> {'
    cfg
  end

  cmd 'show configuration | no-more'
  cmd 'show configuration commands | no-more'

  cfg :telnet do
    username  /login:\s/
    password  /^Password:\s/
  end

  cfg :telnet, :ssh do
    pre_logout 'exit'
  end

end

**Comment by ytti** [comment]
Are you telling me, you cannot give the platform "show configuration" output as configuration?

**Comment by sallysandtrap** [comment]
Yes and no...ill put it in cisco terminology just to show the logic.

If i took the output of "show configuration" pasted it in a file named startup-config and copied it to flash then the vyos router would boot with that configuration. However, i would not be able to copy this output and paste it into the router directly. The "show configuration commands" output is what you need to paste the config back in.

On the backend, pasting a backup of "show configuration commands" in the cli will populate the startup-config file that is read in its raw form when you issue the "show configuration command"

**Comment by ytti** [comment]
I think we need input from author, @thakala

**Comment by acgustafson** [comment]
Both seem to be useful if you had to restore something that was screwed up. One requires you to manually edit /config/config.boot and the other is going through the cli and manually pasting in a command. Having commands you could just paste in seems useful to me.

@ytti since VyOS is like JunOS you would have to prefix everything with set XXX. So currently oxidized backs up the configuration like 
```
system {
        user acgustafson {
            authentication {
                encrypted-password ****************
                plaintext-password ****************
            }
        }
}
```
You can't paste that into VyOS and be live again. You instead have to run these commands which @sallysandtrap is proposing we backup by running `show configuration commands`

```
set system login user acgustafson authentication encrypted-password 'myencryptedpasswordhere'
set system login user acgustafson authentication plaintext-password ''
```

**Comment by chipgwyn** [comment]
I wanted similar functionality and did the following:

`  cmd 'show configuration | no-more' do |cfg|
    cfg
  end

  cmd 'show configuration commands | no-more' do |cfg|
    comment cfg
  end`

This way you end with both versions in the config file.  If need be you can load the config directly since all the "set" commands are commented out.  If you need the set commands they're there at the bottom.  Assuming you'll be copy/pasting those in anyway and its easy to strip comments from the beginning of the line.

**Comment by ytti** [comment]
We do support multifile saves.

```
cmd 'show configuration commands | no-more' do |cfg|
  cfg.type = 'set_commands'
  cfg
end
```

**Comment by yzguy** [comment]
Relevant issue/ticket:
https://github.com/ytti/oxidized/issues/1163
https://github.com/ytti/oxidized/pull/1164

**Comment by wk** [comment]
This appears to be resolved in #1164.
